libstr.h: function strdup already exists on macOS 10.13

Building on macOS 10.13.3 gives a series of errors due to altered definition of the function `strdup`:

```
/.../asar/src/asar/libstr.h:607:15: error: exception specification in declaration
      does not match previous declaration
inline char * strdup(const char * str) throw ()
              ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk/usr/include/string.h:117:7: note:
      previous declaration is here
char    *strdup(const char *__s1);
         ^
/.../asar/src/asar/interface-lib.cpp:72:17: error: call to 'strdup' is ambiguous
        myerr.filename=strdup(thisfilename);
                       ^~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk/usr/include/string.h:117:7: note:
      candidate function
char    *strdup(const char *__s1);
         ^
/.../asar/src/asar/libstr.h:607:15: note: candidate function
inline char * strdup(const char * str) throw ()
              ^
... (followed by lots of other "call to 'strdup' is ambiguous" errors)
6 errors generated.
```

Changing the preprocessor check in `src/asar/libstr.h` from

```c++
#if !defined(_MSC_VER)
inline char * strdup(const char * str) throw ()
```

to

```c++
#if !defined(_MSC_VER) && !defined(__APPLE__)
inline char * strdup(const char * str) throw ()
```

solved the issue for me.

--- a/src/asar/libstr.h
+++ b/src/asar/libstr.h
@@ -603,7 +603,7 @@ inline bool striwcmp(const char *s, const char *p) {
 }
 
 //I thought this one was FreeBSD-only, and now it suddenly exists in MSVC but not GCC?
-#if !defined(_MSC_VER)
+#if !defined(_MSC_VER) && !defined(__APPLE__)
 inline char * strdup(const char * str) throw ()
 {
 	char * a=(char*)malloc(sizeof(char)*(strlen(str)+1));

